examples/stable_diffusion_xl/gm/models/trainer_factory_new.py

```diff
@@ -0,0 +1,77 @@
+# This file only applies to static graph mode
+
+from gm.util import append_dims
+
+import mindspore as ms
+from mindspore import nn, ops
+
+
+class LatentDiffusionWithLoss(nn.Cell):
+    def __init__(self, model):
+        super(LatentDiffusionWithLoss, self).__init__()
+        self.model = model.model
+        self.denoiser = model.denoiser
+        self.loss_fn = model.loss_fn
+
+        # first stage model
+        self.scale_factor = model.scale_factor
+        self.disable_first_stage_amp = model.disable_first_stage_amp
+        self.first_stage_model = model.first_stage_model
+        if self.disable_first_stage_amp:
+            self.first_stage_model.to_float(ms.float32)
+
+        self.sigma_sampler = model.sigma_sampler
+        self.conditioner = model.conditioner
+        self.loss_fn = model.loss_fn
+        self.denoiser = model.denoiser
+
+    def get_first_stage_encoding(self, x):
+        if self.disable_first_stage_amp:
+            x = x.to(ms.float32)
+        else:
+            x = x.to(ms.float16)
+        z = ops.stop_gradient(self.first_stage_model.encode(x))
+        z = self.scale_factor * z
+        return z
+
+    def construct(self, x, *tokens):
+        # get latent target
+        x = self.get_first_stage_encoding(x)
+
+        # get condition
+        vector, crossattn, concat = self.conditioner(*tokens)
+        context, y = crossattn, vector
+
+        # get noise and sigma
+        sigmas = self.sigma_sampler(x.shape[0])
```

Comment:
sigma sampling can be placed outside the graph later

Reply:
But sampling performs ops() on tensors. Is it good to place it outside of `construct`?